prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/migrationtoolkit/GetAnalysisTask.java

```diff
@@ -85,25 +85,29 @@ public WorkReport execute(WorkContext workContext) {
 		Result<TaskGroup> result = mtaClient.get(taskGroupID);
 
 		if (result == null) {
+			taskLogger.logErrorWithSlf4j("MTA client returned empty result with no error.");
 			// unexpected
-			return new DefaultWorkReport(WorkStatus.FAILED, new WorkContext(),
+			return new DefaultWorkReport(WorkStatus.REJECTED, new WorkContext(),
 					new IllegalStateException("MTA client returned empty result with no error."));
 		}
 		else if (result instanceof Result.Failure<TaskGroup> failure) {
-			return new DefaultWorkReport(WorkStatus.FAILED, workContext, failure.t());
+			taskLogger.logErrorWithSlf4j("MTA client returned failed result");
+			return new DefaultWorkReport(WorkStatus.REJECTED, workContext, failure.t());
 		}
 		else if (result instanceof Result.Success<TaskGroup> success) {
 			if ("Ready".equals(success.value().state()) && success.value().tasks() != null
 					&& "Succeeded".equals(success.value().tasks()[0].state())) {
-				String reportURL = "%s/hub/applications/%d/bucket/%s".formatted(serverUrl,
+				String reportURL = "%s/hub/applications/%d/bucket%s".formatted(serverUrl,
```

Comment:
not for today, but perhaps this one can use something like URLBuilder to make sure the URL is valid.

Reply:
yea agree